The job **`aarch64-gnu-llvm-19-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17100122605/job/48494175443) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/48494175443) [(plain)](https://github.com/rust-lang/rust/commit/2b796aae99cbea599ffa11378c18a6c0c30c8bae/checks/48494175443/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [codegen] tests/codegen-llvm/s390x-simd.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-19/bin/FileCheck" "--input-file" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll" "/checkout/tests/codegen-llvm/s390x-simd.rs" "--check-prefix=CHECK" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen-llvm/s390x-simd.rs:50:11: error: CHECK: expected string not found in input
// CHECK: call <16 x i8> @llvm.smax.v16i8(<16 x i8> %a, <16 x i8> %b)
          ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:51:55: note: scanning from here
define <16 x i8> @max_i8x16(<16 x i8> %a, <16 x i8> %b) unnamed_addr #1 {
                                                      ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:58:3: note: possible intended match here
 store <16 x i8> %a, ptr %1, align 8
  ^
/checkout/tests/codegen-llvm/s390x-simd.rs:59:11: error: CHECK: expected string not found in input
// CHECK: call <8 x i16> @llvm.smax.v8i16(<8 x i16> %a, <8 x i16> %b)
          ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:71:55: note: scanning from here
define <8 x i16> @max_i16x8(<8 x i16> %a, <8 x i16> %b) unnamed_addr #1 {
                                                      ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:83:2: note: possible intended match here
 call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %1)
 ^
/checkout/tests/codegen-llvm/s390x-simd.rs:68:11: error: CHECK: expected string not found in input
// CHECK: call <4 x i32> @llvm.smax.v4i32(<4 x i32> %a, <4 x i32> %b)
          ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:91:55: note: scanning from here
define <4 x i32> @max_i32x4(<4 x i32> %a, <4 x i32> %b) unnamed_addr #1 {
                                                      ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:98:3: note: possible intended match here
 store <4 x i32> %a, ptr %1, align 8
  ^
/checkout/tests/codegen-llvm/s390x-simd.rs:77:11: error: CHECK: expected string not found in input
// CHECK: call <2 x i64> @llvm.smax.v2i64(<2 x i64> %a, <2 x i64> %b)
          ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:111:55: note: scanning from here
define <2 x i64> @max_i64x2(<2 x i64> %a, <2 x i64> %b) unnamed_addr #1 {
                                                      ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:123:2: note: possible intended match here
 call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %1)
 ^
/checkout/tests/codegen-llvm/s390x-simd.rs:111:12: error: CHECK: expected string not found in input
 // CHECK: call <16 x i8> @llvm.smax.v16i8
           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:145:168: note: scanning from here
define void @max_wrapper_i8x16(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 8 dereferenceable(16) %_0, <16 x i8> %0, <16 x i8> %1) unnamed_addr #2 {
                                                                                                                                                                       ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:147:11: note: possible intended match here
 %2 = alloca [16 x i8], align 8
          ^
/checkout/tests/codegen-llvm/s390x-simd.rs:125:12: error: CHECK: expected string not found in input
 // CHECK: call <2 x i64> @llvm.smax.v2i64
           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:161:168: note: scanning from here
define void @max_wrapper_i64x2(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 8 dereferenceable(16) %_0, <16 x i8> %0, <16 x i8> %1) unnamed_addr #2 {
                                                                                                                                                                       ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll:165:2: note: possible intended match here
 call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %3)
 ^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/s390x-simd/s390x-simd.ll
Check file: /checkout/tests/codegen-llvm/s390x-simd.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
             1: ; ModuleID = 's390x_simd.d93fc7ac282ccdeb-cgu.0' 
             2: source_filename = "s390x_simd.d93fc7ac282ccdeb-cgu.0" 
             3: target datalayout = "E-m:e-i1:8:16-i8:8:16-i64:64-f128:64-v128:64-a:8:16-n32:64" 
             4: target triple = "s390x-unknown-linux-gnu" 
             5:  
             6: ; s390x_simd::simd_max 
             7: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable 
             8: define internal fastcc void @_ZN10s390x_simd8simd_max17h07d61dfd7e36514bE(ptr dead_on_unwind noalias nocapture noundef writable writeonly align 8 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 8 dereferenceable(16) %a, ptr noalias nocapture noundef readonly align 8 dereferenceable(16) %b) unnamed_addr #0 { 
             9: start: 
            10:  %0 = load <2 x i64>, ptr %a, align 8 
            11:  %1 = load <2 x i64>, ptr %b, align 8 
            12:  %2 = tail call <2 x i64> @llvm.smax.v2i64(<2 x i64> %0, <2 x i64> %1) 
            13:  store <2 x i64> %2, ptr %_0, align 8 
            14:  ret void 
            15: } 
            16:  
            17: ; s390x_simd::simd_max 
            18: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable 
            19: define internal fastcc void @_ZN10s390x_simd8simd_max17h2e5a8c4497b89ae1E(ptr dead_on_unwind noalias nocapture noundef writable writeonly align 8 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 8 dereferenceable(16) %a, ptr noalias nocapture noundef readonly align 8 dereferenceable(16) %b) unnamed_addr #0 { 
            20: start: 
            21:  %0 = load <16 x i8>, ptr %a, align 8 
            22:  %1 = load <16 x i8>, ptr %b, align 8 
            23:  %2 = tail call <16 x i8> @llvm.smax.v16i8(<16 x i8> %0, <16 x i8> %1) 
            24:  store <16 x i8> %2, ptr %_0, align 8 
            25:  ret void 
            26: } 
            27:  
            28: ; s390x_simd::simd_max 
            29: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable 
            30: define internal fastcc void @_ZN10s390x_simd8simd_max17h3f7ac6eb87c39a7cE(ptr dead_on_unwind noalias nocapture noundef writable writeonly align 8 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 8 dereferenceable(16) %a, ptr noalias nocapture noundef readonly align 8 dereferenceable(16) %b) unnamed_addr #0 { 
            31: start: 
            32:  %0 = load <8 x i16>, ptr %a, align 8 
            33:  %1 = load <8 x i16>, ptr %b, align 8 
            34:  %2 = tail call <8 x i16> @llvm.smax.v8i16(<8 x i16> %0, <8 x i16> %1) 
            35:  store <8 x i16> %2, ptr %_0, align 8 
            36:  ret void 
            37: } 
            38:  
            39: ; s390x_simd::simd_max 
            40: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable 
            41: define internal fastcc void @_ZN10s390x_simd8simd_max17h8f1cd44bd8ffe3e8E(ptr dead_on_unwind noalias nocapture noundef writable writeonly align 8 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 8 dereferenceable(16) %a, ptr noalias nocapture noundef readonly align 8 dereferenceable(16) %b) unnamed_addr #0 { 
            42: start: 
            43:  %0 = load <4 x i32>, ptr %a, align 8 
            44:  %1 = load <4 x i32>, ptr %b, align 8 
            45:  %2 = tail call <4 x i32> @llvm.smax.v4i32(<4 x i32> %0, <4 x i32> %1) 
            46:  store <4 x i32> %2, ptr %_0, align 8 
            47:  ret void 
            48: } 
            49:  
            50: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable 
            51: define <16 x i8> @max_i8x16(<16 x i8> %a, <16 x i8> %b) unnamed_addr #1 { 
check:50'0                                                            X~~~~~~~~~~~~~~~~~~~ error: no match found
            52: start: 
check:50'0      ~~~~~~~
            53:  %0 = alloca [16 x i8], align 8 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            54:  %1 = alloca [16 x i8], align 8 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            55:  %2 = alloca [16 x i8], align 8 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            56:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %2) 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            57:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %1) 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            58:  store <16 x i8> %a, ptr %1, align 8 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:50'1        ?                                   possible intended match
            59:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %0) 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            60:  store <16 x i8> %b, ptr %0, align 8 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            61: ; call s390x_simd::simd_max 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            62:  call fastcc void @_ZN10s390x_simd8simd_max17h2e5a8c4497b89ae1E(ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %2, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %1, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %0) #6 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            63:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %1) 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            64:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %0) 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            65:  %_0 = load <16 x i8>, ptr %2, align 8 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            66:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %2) 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            67:  ret <16 x i8> %_0 
check:50'0      ~~~~~~~~~~~~~~~~~~~
            68: } 
check:50'0      ~~
            69:  
check:50'0      ~
            70: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            71: define <8 x i16> @max_i16x8(<8 x i16> %a, <8 x i16> %b) unnamed_addr #1 { 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:59'0                                                            X~~~~~~~~~~~~~~~~~~~ error: no match found
            72: start: 
check:59'0      ~~~~~~~
            73:  %0 = alloca [16 x i8], align 8 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            74:  %1 = alloca [16 x i8], align 8 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            75:  %2 = alloca [16 x i8], align 8 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            76:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %2) 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            77:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %1) 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            78:  store <8 x i16> %a, ptr %1, align 8 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            79:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %0) 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            80:  store <8 x i16> %b, ptr %0, align 8 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            81: ; call s390x_simd::simd_max 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            82:  call fastcc void @_ZN10s390x_simd8simd_max17h3f7ac6eb87c39a7cE(ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %2, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %1, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %0) #6 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            83:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %1) 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:59'1       ?                                                        possible intended match
            84:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %0) 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            85:  %_0 = load <8 x i16>, ptr %2, align 8 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            86:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %2) 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            87:  ret <8 x i16> %_0 
check:59'0      ~~~~~~~~~~~~~~~~~~~
            88: } 
check:59'0      ~~
            89:  
check:59'0      ~
            90: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            91: define <4 x i32> @max_i32x4(<4 x i32> %a, <4 x i32> %b) unnamed_addr #1 { 
check:59'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:68'0                                                            X~~~~~~~~~~~~~~~~~~~ error: no match found
            92: start: 
check:68'0      ~~~~~~~
            93:  %0 = alloca [16 x i8], align 8 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            94:  %1 = alloca [16 x i8], align 8 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            95:  %2 = alloca [16 x i8], align 8 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            96:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %2) 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            97:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %1) 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            98:  store <4 x i32> %a, ptr %1, align 8 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:68'1        ?                                   possible intended match
            99:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %0) 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           100:  store <4 x i32> %b, ptr %0, align 8 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           101: ; call s390x_simd::simd_max 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           102:  call fastcc void @_ZN10s390x_simd8simd_max17h8f1cd44bd8ffe3e8E(ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %2, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %1, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %0) #6 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           103:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %1) 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           104:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %0) 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           105:  %_0 = load <4 x i32>, ptr %2, align 8 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           106:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %2) 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           107:  ret <4 x i32> %_0 
check:68'0      ~~~~~~~~~~~~~~~~~~~
           108: } 
check:68'0      ~~
           109:  
check:68'0      ~
           110: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           111: define <2 x i64> @max_i64x2(<2 x i64> %a, <2 x i64> %b) unnamed_addr #1 { 
check:68'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:77'0                                                            X~~~~~~~~~~~~~~~~~~~ error: no match found
           112: start: 
check:77'0      ~~~~~~~
           113:  %0 = alloca [16 x i8], align 8 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           114:  %1 = alloca [16 x i8], align 8 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           115:  %2 = alloca [16 x i8], align 8 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           116:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %2) 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           117:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %1) 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           118:  store <2 x i64> %a, ptr %1, align 8 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           119:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %0) 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           120:  store <2 x i64> %b, ptr %0, align 8 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           121: ; call s390x_simd::simd_max 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           122:  call fastcc void @_ZN10s390x_simd8simd_max17h07d61dfd7e36514bE(ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %2, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %1, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %0) #6 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           123:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %1) 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:77'1       ?                                                        possible intended match
           124:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %0) 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           125:  %_0 = load <2 x i64>, ptr %2, align 8 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           126:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %2) 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           127:  ret <2 x i64> %_0 
check:77'0      ~~~~~~~~~~~~~~~~~~~
           128: } 
check:77'0      ~~
           129:  
check:77'0      ~
           130: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           131: define <4 x float> @choose_f32x4(<4 x float> %a, <4 x float> %b, i1 noundef zeroext %c) unnamed_addr #1 { 
check:77'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           132: start: 
           133:  %a.b = select i1 %c, <4 x float> %a, <4 x float> %b 
           134:  ret <4 x float> %a.b 
           135: } 
           136:  
           137: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable 
           138: define <2 x double> @choose_f64x2(<2 x double> %a, <2 x double> %b, i1 noundef zeroext %c) unnamed_addr #1 { 
           139: start: 
           140:  %a.b = select i1 %c, <2 x double> %a, <2 x double> %b 
           141:  ret <2 x double> %a.b 
           142: } 
           143:  
           144: ; Function Attrs: mustprogress nofree noinline norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable 
           145: define void @max_wrapper_i8x16(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 8 dereferenceable(16) %_0, <16 x i8> %0, <16 x i8> %1) unnamed_addr #2 { 
check:111'0                                                                                                                                                                            X~~~~~~~~~~~~~~~~~~~~~~ error: no match found
           146: start: 
check:111'0     ~~~~~~~
           147:  %2 = alloca [16 x i8], align 8 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:111'1               ?                      possible intended match
           148:  %3 = alloca [16 x i8], align 8 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           149:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %3) 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           150:  store <16 x i8> %0, ptr %3, align 8 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           151:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %2) 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           152:  store <16 x i8> %1, ptr %2, align 8 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           153: ; call s390x_simd::simd_max 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           154:  call fastcc void @_ZN10s390x_simd8simd_max17h2e5a8c4497b89ae1E(ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %_0, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %3, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %2) #6 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           155:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %3) 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           156:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %2) 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           157:  ret void 
check:111'0     ~~~~~~~~~~
           158: } 
check:111'0     ~~
           159:  
check:111'0     ~
           160: ; Function Attrs: mustprogress nofree noinline norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           161: define void @max_wrapper_i64x2(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 8 dereferenceable(16) %_0, <16 x i8> %0, <16 x i8> %1) unnamed_addr #2 { 
check:111'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:125'0                                                                                                                                                                            X~~~~~~~~~~~~~~~~~~~~~~ error: no match found
           162: start: 
check:125'0     ~~~~~~~
           163:  %2 = alloca [16 x i8], align 8 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           164:  %3 = alloca [16 x i8], align 8 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           165:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %3) 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:125'1      ?                                                          possible intended match
           166:  store <16 x i8> %0, ptr %3, align 8 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           167:  call void @llvm.lifetime.start.p0(i64 16, ptr nonnull %2) 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           168:  store <16 x i8> %1, ptr %2, align 8 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           169: ; call s390x_simd::simd_max 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           170:  call fastcc void @_ZN10s390x_simd8simd_max17h07d61dfd7e36514bE(ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %_0, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %3, ptr noalias nocapture noundef nonnull align 8 dereferenceable(16) %2) #6 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           171:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %3) 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           172:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %2) 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           173:  ret void 
check:125'0     ~~~~~~~~~~
           174: } 
check:125'0     ~~
           175:  
check:125'0     ~
           176: ; Function Attrs: mustprogress nofree noinline norecurse nosync nounwind willreturn memory(argmem: write) uwtable 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           177: define void @choose_wrapper_f64x2(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 8 dereferenceable(16) %_0, <16 x i8> %0, <16 x i8> %1, i1 noundef zeroext %c) unnamed_addr #3 { 
check:125'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           178: start: 
           179:  %a.b.i.v = select i1 %c, <16 x i8> %0, <16 x i8> %1 
           180:  store <16 x i8> %a.b.i.v, ptr %_0, align 8 
           181:  ret void 
           182: } 
           183:  
           184: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
           185: declare void @llvm.lifetime.start.p0(i64 immarg, ptr nocapture) #4 
           186:  
           187: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
           188: declare void @llvm.lifetime.end.p0(i64 immarg, ptr nocapture) #4 
           189:  
           190: ; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none) 
           191: declare <2 x i64> @llvm.smax.v2i64(<2 x i64>, <2 x i64>) #5 
           192:  
           193: ; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none) 
           194: declare <16 x i8> @llvm.smax.v16i8(<16 x i8>, <16 x i8>) #5 
           195:  
           196: ; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none) 
           197: declare <8 x i16> @llvm.smax.v8i16(<8 x i16>, <8 x i16>) #5 
           198:  
           199: ; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none) 
           200: declare <4 x i32> @llvm.smax.v4i32(<4 x i32>, <4 x i32>) #5 
           201:  
           202: attributes #0 = { mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable "probe-stack"="inline-asm" "target-cpu"="z10" } 
           203: attributes #1 = { mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable "probe-stack"="inline-asm" "target-cpu"="z10" "target-features"="+vector" } 
           204: attributes #2 = { mustprogress nofree noinline norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable "probe-stack"="inline-asm" "target-cpu"="z10" "target-features"="+vector" } 
           205: attributes #3 = { mustprogress nofree noinline norecurse nosync nounwind willreturn memory(argmem: write) uwtable "probe-stack"="inline-asm" "target-cpu"="z10" "target-features"="+vector" } 
           206: attributes #4 = { mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) } 
           207: attributes #5 = { nocallback nofree nosync nounwind speculatable willreturn memory(none) } 
           208: attributes #6 = { nounwind } 
           209:  
           210: !llvm.module.flags = !{!0} 
           211: !llvm.ident = !{!1} 
           212:  
           213: !0 = !{i32 8, !"PIC Level", i32 2} 
           214: !1 = !{!"rustc version 1.91.0-nightly (678560a31 2025-08-20)"} 
>>>>>>
------------------------------------------



```

</details>

--- a/tests/assembly-llvm/s390x-vector-abi.rs
+++ b/tests/assembly-llvm/s390x-vector-abi.rs
@@ -1,4 +1,5 @@
 //@ revisions: z10 z10_vector z13 z13_no_vector
+//@ add-core-stubs
 // ignore-tidy-linelength
 //@ assembly-output: emit-asm
 //@ compile-flags: -Copt-level=3 -Z merge-functions=disabled
@@ -18,24 +19,8 @@
 // Cases where vector feature is disabled are rejected.
 // See tests/ui/simd-abi-checks-s390x.rs for test for them.
 
-#[lang = "pointee_sized"]
-pub trait PointeeSized {}
-
-#[lang = "meta_sized"]
-pub trait MetaSized: PointeeSized {}
-
-#[lang = "sized"]
-pub trait Sized: MetaSized {}
-#[lang = "copy"]
-pub trait Copy {}
-#[lang = "freeze"]
-pub trait Freeze {}
-
-impl<T: Copy, const N: usize> Copy for [T; N] {}
-
-#[lang = "phantom_data"]
-pub struct PhantomData<T: ?Sized>;
-impl<T: ?Sized> Copy for PhantomData<T> {}
+extern crate minicore;
+use minicore::*;
 
 #[repr(simd)]
 pub struct i8x8([i8; 8]);
@@ -52,8 +37,6 @@ pub struct WrapperWithZst<T>(T, PhantomData<()>);
 #[repr(transparent)]
 pub struct TransparentWrapper<T>(T);
 
-impl Copy for i8 {}
-impl Copy for i64 {}
 impl Copy for i8x8 {}
 impl Copy for i8x16 {}
 impl Copy for i8x32 {}
@@ -221,7 +204,7 @@ unsafe extern "C" fn vector_transparent_wrapper_ret_large(
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_arg_small(x: i8x8) -> i64 {
-    unsafe { *(&x as *const i8x8 as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 // CHECK-LABEL: vector_arg:
 // CHECK: vlgvg %r2, %v24, 0
@@ -229,7 +212,7 @@ unsafe extern "C" fn vector_arg_small(x: i8x8) -> i64 {
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_arg(x: i8x16) -> i64 {
-    unsafe { *(&x as *const i8x16 as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 // CHECK-LABEL: vector_arg_large:
 // CHECK: lg %r2, 0(%r2)
@@ -237,7 +220,7 @@ unsafe extern "C" fn vector_arg(x: i8x16) -> i64 {
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_arg_large(x: i8x32) -> i64 {
-    unsafe { *(&x as *const i8x32 as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 
 // CHECK-LABEL: vector_wrapper_arg_small:
@@ -246,7 +229,7 @@ unsafe extern "C" fn vector_arg_large(x: i8x32) -> i64 {
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_wrapper_arg_small(x: Wrapper<i8x8>) -> i64 {
-    unsafe { *(&x as *const Wrapper<i8x8> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 // CHECK-LABEL: vector_wrapper_arg:
 // CHECK: vlgvg %r2, %v24, 0
@@ -254,7 +237,7 @@ unsafe extern "C" fn vector_wrapper_arg_small(x: Wrapper<i8x8>) -> i64 {
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_wrapper_arg(x: Wrapper<i8x16>) -> i64 {
-    unsafe { *(&x as *const Wrapper<i8x16> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 // CHECK-LABEL: vector_wrapper_arg_large:
 // CHECK: lg %r2, 0(%r2)
@@ -262,7 +245,7 @@ unsafe extern "C" fn vector_wrapper_arg(x: Wrapper<i8x16>) -> i64 {
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_wrapper_arg_large(x: Wrapper<i8x32>) -> i64 {
-    unsafe { *(&x as *const Wrapper<i8x32> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 
 // https://github.com/rust-lang/rust/pull/131586#discussion_r1837071121
@@ -272,7 +255,7 @@ unsafe extern "C" fn vector_wrapper_arg_large(x: Wrapper<i8x32>) -> i64 {
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_wrapper_padding_arg(x: WrapperAlign16<i8x8>) -> i64 {
-    unsafe { *(&x as *const WrapperAlign16<i8x8> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 
 // CHECK-LABEL: vector_wrapper_with_zst_arg_small:
@@ -282,7 +265,7 @@ unsafe extern "C" fn vector_wrapper_padding_arg(x: WrapperAlign16<i8x8>) -> i64 
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_wrapper_with_zst_arg_small(x: WrapperWithZst<i8x8>) -> i64 {
-    unsafe { *(&x as *const WrapperWithZst<i8x8> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 // CHECK-LABEL: vector_wrapper_with_zst_arg:
 // CHECK: lg %r2, 0(%r2)
@@ -290,7 +273,7 @@ unsafe extern "C" fn vector_wrapper_with_zst_arg_small(x: WrapperWithZst<i8x8>) 
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_wrapper_with_zst_arg(x: WrapperWithZst<i8x16>) -> i64 {
-    unsafe { *(&x as *const WrapperWithZst<i8x16> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 // CHECK-LABEL: vector_wrapper_with_zst_arg_large:
 // CHECK: lg %r2, 0(%r2)
@@ -298,7 +281,7 @@ unsafe extern "C" fn vector_wrapper_with_zst_arg(x: WrapperWithZst<i8x16>) -> i6
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_wrapper_with_zst_arg_large(x: WrapperWithZst<i8x32>) -> i64 {
-    unsafe { *(&x as *const WrapperWithZst<i8x32> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 
 // CHECK-LABEL: vector_transparent_wrapper_arg_small:
@@ -307,7 +290,7 @@ unsafe extern "C" fn vector_wrapper_with_zst_arg_large(x: WrapperWithZst<i8x32>)
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_transparent_wrapper_arg_small(x: TransparentWrapper<i8x8>) -> i64 {
-    unsafe { *(&x as *const TransparentWrapper<i8x8> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 // CHECK-LABEL: vector_transparent_wrapper_arg:
 // CHECK: vlgvg %r2, %v24, 0
@@ -315,7 +298,7 @@ unsafe extern "C" fn vector_transparent_wrapper_arg_small(x: TransparentWrapper<
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_transparent_wrapper_arg(x: TransparentWrapper<i8x16>) -> i64 {
-    unsafe { *(&x as *const TransparentWrapper<i8x16> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 // CHECK-LABEL: vector_transparent_wrapper_arg_large:
 // CHECK: lg %r2, 0(%r2)
@@ -323,5 +306,5 @@ unsafe extern "C" fn vector_transparent_wrapper_arg(x: TransparentWrapper<i8x16>
 #[cfg_attr(no_vector, target_feature(enable = "vector"))]
 #[no_mangle]
 unsafe extern "C" fn vector_transparent_wrapper_arg_large(x: TransparentWrapper<i8x32>) -> i64 {
-    unsafe { *(&x as *const TransparentWrapper<i8x32> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }

--- a/tests/codegen-llvm/s390x-simd.rs
+++ b/tests/codegen-llvm/s390x-simd.rs
@@ -6,7 +6,7 @@
 
 #![crate_type = "rlib"]
 #![feature(no_core, asm_experimental_arch)]
-#![feature(s390x_target_feature, simd_ffi, link_llvm_intrinsics, repr_simd)]
+#![feature(s390x_target_feature, simd_ffi, intrinsics, repr_simd)]
 #![no_core]
 
 extern crate minicore;
@@ -30,16 +30,20 @@ struct f32x4([f32; 4]);
 #[repr(simd)]
 struct f64x2([f64; 2]);
 
-#[allow(improper_ctypes)]
-extern "C" {
-    #[link_name = "llvm.smax.v16i8"]
-    fn vmxb(a: i8x16, b: i8x16) -> i8x16;
-    #[link_name = "llvm.smax.v8i16"]
-    fn vmxh(a: i16x8, b: i16x8) -> i16x8;
-    #[link_name = "llvm.smax.v4i32"]
-    fn vmxf(a: i32x4, b: i32x4) -> i32x4;
-    #[link_name = "llvm.smax.v2i64"]
-    fn vmxg(a: i64x2, b: i64x2) -> i64x2;
+impl Copy for i8x16 {}
+impl Copy for i16x8 {}
+impl Copy for i32x4 {}
+impl Copy for i64x2 {}
+
+#[rustc_intrinsic]
+unsafe fn simd_ge<T, U>(x: T, y: T) -> U;
+
+#[rustc_intrinsic]
+unsafe fn simd_select<M, V>(mask: M, a: V, b: V) -> V;
+
+#[inline(always)]
+unsafe fn simd_max<T: Copy>(a: T, b: T) -> T {
+    simd_select(simd_ge::<T, T>(a, b), a, b)
 }
 
 // CHECK-LABEL: define <16 x i8> @max_i8x16
@@ -48,7 +52,7 @@ extern "C" {
 #[no_mangle]
 #[target_feature(enable = "vector")]
 pub unsafe extern "C" fn max_i8x16(a: i8x16, b: i8x16) -> i8x16 {
-    vmxb(a, b)
+    simd_max(a, b)
 }
 
 // CHECK-LABEL: define <8 x i16> @max_i16x8
@@ -57,7 +61,7 @@ pub unsafe extern "C" fn max_i8x16(a: i8x16, b: i8x16) -> i8x16 {
 #[no_mangle]
 #[target_feature(enable = "vector")]
 pub unsafe extern "C" fn max_i16x8(a: i16x8, b: i16x8) -> i16x8 {
-    vmxh(a, b)
+    simd_max(a, b)
 }
 
 // CHECK-LABEL: define <4 x i32> @max_i32x4
@@ -66,7 +70,7 @@ pub unsafe extern "C" fn max_i16x8(a: i16x8, b: i16x8) -> i16x8 {
 #[no_mangle]
 #[target_feature(enable = "vector")]
 pub unsafe extern "C" fn max_i32x4(a: i32x4, b: i32x4) -> i32x4 {
-    vmxf(a, b)
+    simd_max(a, b)
 }
 
 // CHECK-LABEL: define <2 x i64> @max_i64x2
@@ -75,7 +79,7 @@ pub unsafe extern "C" fn max_i32x4(a: i32x4, b: i32x4) -> i32x4 {
 #[no_mangle]
 #[target_feature(enable = "vector")]
 pub unsafe extern "C" fn max_i64x2(a: i64x2, b: i64x2) -> i64x2 {
-    vmxg(a, b)
+    simd_max(a, b)
 }
 
 // CHECK-LABEL: define <4 x float> @choose_f32x4
@@ -108,7 +112,7 @@ pub unsafe extern "C" fn max_wrapper_i8x16(a: Wrapper<i8x16>, b: Wrapper<i8x16>)
     // CHECK: call <16 x i8> @llvm.smax.v16i8
     // CHECK-SAME: <16 x i8>
     // CHECK-SAME: <16 x i8>
-    Wrapper(vmxb(a.0, b.0))
+    Wrapper(simd_max(a.0, b.0))
 }
 
 #[no_mangle]
@@ -122,7 +126,7 @@ pub unsafe extern "C" fn max_wrapper_i64x2(a: Wrapper<i64x2>, b: Wrapper<i64x2>)
     // CHECK: call <2 x i64> @llvm.smax.v2i64
     // CHECK-SAME: <2 x i64>
     // CHECK-SAME: <2 x i64>
-    Wrapper(vmxg(a.0, b.0))
+    Wrapper(simd_max(a.0, b.0))
 }
 
 #[no_mangle]

--- a/tests/ui/abi/simd-abi-checks-s390x.rs
+++ b/tests/ui/abi/simd-abi-checks-s390x.rs
@@ -110,47 +110,47 @@ extern "C" fn vector_transparent_wrapper_ret_large(
 #[no_mangle]
 extern "C" fn vector_arg_small(x: i8x8) -> i64 {
     //~^ ERROR requires the `vector` target feature, which is not enabled
-    unsafe { *(&x as *const i8x8 as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 #[no_mangle]
 extern "C" fn vector_arg(x: i8x16) -> i64 {
     //~^ ERROR requires the `vector` target feature, which is not enabled
-    unsafe { *(&x as *const i8x16 as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 #[no_mangle]
 extern "C" fn vector_arg_large(x: i8x32) -> i64 {
     // Ok
-    unsafe { *(&x as *const i8x32 as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 
 #[no_mangle]
 extern "C" fn vector_wrapper_arg_small(x: Wrapper<i8x8>) -> i64 {
     //~^ ERROR requires the `vector` target feature, which is not enabled
-    unsafe { *(&x as *const Wrapper<i8x8> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 #[no_mangle]
 extern "C" fn vector_wrapper_arg(x: Wrapper<i8x16>) -> i64 {
     //~^ ERROR requires the `vector` target feature, which is not enabled
-    unsafe { *(&x as *const Wrapper<i8x16> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 #[no_mangle]
 extern "C" fn vector_wrapper_arg_large(x: Wrapper<i8x32>) -> i64 {
     // Ok
-    unsafe { *(&x as *const Wrapper<i8x32> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 
 #[no_mangle]
 extern "C" fn vector_transparent_wrapper_arg_small(x: TransparentWrapper<i8x8>) -> i64 {
     //~^ ERROR requires the `vector` target feature, which is not enabled
-    unsafe { *(&x as *const TransparentWrapper<i8x8> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 #[no_mangle]
 extern "C" fn vector_transparent_wrapper_arg(x: TransparentWrapper<i8x16>) -> i64 {
     //~^ ERROR requires the `vector` target feature, which is not enabled
-    unsafe { *(&x as *const TransparentWrapper<i8x16> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }
 #[no_mangle]
 extern "C" fn vector_transparent_wrapper_arg_large(x: TransparentWrapper<i8x32>) -> i64 {
     // Ok
-    unsafe { *(&x as *const TransparentWrapper<i8x32> as *const i64) }
+    unsafe { *(&raw const x as *const i64) }
 }

